crypto/ube/ube_test.cc

```diff
@@ -94,18 +174,56 @@ TEST(Ube, MockedMethodTests) {
 
   // Now try to increment both fork and snapsafe generation numbers. We expect
   // to see one increment in the ube generation number and then stability.
-  set_snapsafe_generation_number_FOR_TESTING(20);
   set_fork_generation_number_FOR_TESTING(20);
+  set_snapsafe_generation_number_FOR_TESTING(20);
+  ASSERT_TRUE(CRYPTO_get_ube_generation_number(&generation_number));
 
-  // Check that ube generation number incremented once.
+  // And that it's now stable.
+  cached_generation_number = generation_number;
+  generation_number = 0;
+  ASSERT_TRUE(CRYPTO_get_ube_generation_number(&generation_number));
+  ASSERT_EQ(generation_number, cached_generation_number);
+
+  // Increment both, this time using higher increments than 1 and different
+  // increment values.
+  set_fork_generation_number_FOR_TESTING(24);
+  set_snapsafe_generation_number_FOR_TESTING(23);
+
+  // Generation number should have incremented once.
   cached_generation_number = generation_number;
   generation_number = 0;
   ASSERT_TRUE(CRYPTO_get_ube_generation_number(&generation_number));
   ASSERT_EQ(generation_number, cached_generation_number + 1);
 
-  // And that it's now stable.
+  // Should be stable again.
   cached_generation_number = generation_number;
   generation_number = 0;
   ASSERT_TRUE(CRYPTO_get_ube_generation_number(&generation_number));
   ASSERT_EQ(generation_number, cached_generation_number);
+
+  // Try strictly smaller values. These are values previously used as fork and
```

Comment:
This test is getting really long and it's hard to check if you're covering all possible combinations of snapsafe/fork incrementing/decrementing. Can you break this up and/or make this a parametrized test?

Reply:
True, split them up into: snap, fork, both.
Could do parametrised, but little value right now I think..

Reply:
I think there is value doing it now. It's currently hard to tell if you have them all covered.

Reply:
Refactored into common parts